### ISSUE TITLE
Can't use :config-in-comment with the :unused-referred-var linter

### ISSUE THREAD

**Issue by joshgelbard** [clj-kondo/clj-kondo#2087]
**version**

v2023.05.18

**platform**

native, macos-aarch64

**problem**

`:config-in-comment` ignores my attempts to turn off the `:unused-referred-var` linter.

**minimal repro**
```clojure
(ns foo) ;; foo.clj
(comment (require '[a :refer [shouldnt-warn]]) a/a)
```
```edn
;; config.edn
{:config-in-comment {:linters {:unused-referred-var {:level :off}}}}
```
```shell
$ clj-kondo --config config.edn --lint foo.clj
```

**brief debug attempts repro**
While trying to figure out if I was just holding it wrong, I tried swapping out `:config-in-comment` for `{:config-in-call {clojure.core/comment ...}}` and testing with `:ignore` instead of `{:level :off}`.
```clojure
(ns bar) ;; bar.clj
(require '[clojure.set
           :refer [union]])
(= 1)
(comment
  (require '[clojure.string
             :refer [split]])
  (< 2))

```

```edn
;; config-1.edn - config-4.edn
{:config-in-comment {:linters {:unused-namespace {:level :off} :unused-referred-var {:level :off} :single-operand-comparison {:level :off}}}}
{:config-in-comment {:ignore [:unused-namespace :unused-referred-var :single-operand-comparison]}}
{:config-in-call {clojure.core/comment {:ignore [:unused-namespace :unused-referred-var :single-operand-comparison]}}}
{:config-in-call {clojure.core/comment {:linters {:unused-namespace {:level :off} :unused-referred-var {:level :off} :single-operand-comparison {:level :off}}}}}
```
  ```shell
  for c in config-{1..4}.edn; do printf "\n\n$c:\n" && clj-kondo --config $c --lint bar.clj; done
  ```

**expected behavior**

In the one-liner minimal repro, no warnings should be printed.

In each of the debug repros, lines 2, 3, and 4 should print warnings, while nothing should be printed for lines 6, 7, or 8.

**actual behavior**

In the minimal repro, `warning: #'a/shouldnt-warn is referred but never used` is printed.

In the debug repro, configs 1, 3, and 4 print `warning: #'clojure.string/split is referred but never used` for line 7. config-2.edn prints an unused namespace warning for line 6 and a single-operand-comparison warning for line 8, too, which the others don't.
